### ISSUE TITLE
MS3 Libraries may be missing fragments in NIST parsing b/c of isMzAlreadyExists check

### ISSUE THREAD

**Issue by PMSeitzer** [eugenemel/maven#237]
in `database.cpp`:
```
                     bool isMzAlreadyExists = false;
                     if (cpd->fragment_mzs.size() > 0) {
                         float prevMz = cpd->fragment_mzs.at(cpd->fragment_mzs.size()-1);
                         if (abs(prevMz-mz) < 1e-6) {
                             isMzAlreadyExists = true;
                         }
                     }

                     if (!isMzAlreadyExists) {
                         cpd->fragment_intensity.push_back(ints);
                         cpd->fragment_mzs.push_back(mz);
                         int frag_indx = cpd->fragment_mzs.size()-1;

                         if(mzintpair.size() >= 3) {
                             cpd->fragment_iontype[frag_indx] = mzintpair.at(2).toStdString();
                             QString fragLabel("");
                             for (unsigned int k = 2; k < mzintpair.size(); k++) {
                                 if (k > 2) {
                                     fragLabel.append(" ");
                                 }
                                 fragLabel.append(mzintpair.at(k));
                             }
                             cpd->fragment_labels.push_back(fragLabel.toStdString());
                        } else {
                             cpd->fragment_labels.push_back("");
                        }
                     }
```

To test this, need to write some tests

**Comment by PMSeitzer** [comment]
This is a real problem. Also need to look for duplicate`m/zs` that may not be the previous `m/z`.

**Comment by PMSeitzer** [comment]
Even for matches that agree between `maven` and `mzkitcpp`, the fragment labels are incorrect:

![MAVEN_TG_36_1_screenshot](https://user-images.githubusercontent.com/1757701/88211452-95039280-cc0a-11ea-939d-ae94421bc1bb.png)

![mzkitcpp_TG_36_1_screenshot](https://user-images.githubusercontent.com/1757701/88211459-98971980-cc0a-11ea-9e51-e00c037b98e3.png)

**Comment by PMSeitzer** [comment]
Another example of match label disagreement between `maven` and `mzkitcpp`:
![MAVEN_TG_18_1_20_5_22_6](https://user-images.githubusercontent.com/1757701/88211784-0b07f980-cc0b-11ea-832f-1f1813305f83.png)
![mzkitcpp_TG_18_1_20_5_22_6](https://user-images.githubusercontent.com/1757701/88211793-0d6a5380-cc0b-11ea-9754-da5915a307ec.png)

**Comment by PMSeitzer** [comment]
One problem:

1. start from empty file
2. load data, TG lib
3. search
4. save results, including some summarized compounds
5. open file
6. load data, TG lib
7. search again

The results differ, b/c the summarized compounds saved from `4` and reopened from `5` are added to the compounds database. Instead, only the specified library should be searched.

**Comment by PMSeitzer** [comment]
This has been implemented, and is basically good enough for now.  Still no check on merging `MS3` fragments, this is the responsibility of the library generator.

**Comment by PMSeitzer** [comment]
The match label disagreements described here were solved in the completion of #238
